core/src/apps/common/cbor.py

```diff
@@ -78,9 +79,17 @@ def _cbor_encode(value: Value) -> Iterable[bytes]:
         for k, v in sorted_map:
             yield k
             yield from _cbor_encode(v)
+    elif isinstance(value, Set):
+        sorted_set = sorted(set(encode(v) for v in value.array))
```

Comment:
The best way I could come up with to deal with duplicates. I create a set from the encoded data, which then gets sorted. I can't use a set from from the start, because lists aren't hashable and thus can't be added to a set.

Reply:
I don't see what the connection to IndefiniteLengthArray is.